homeassistant/const.py

```diff
@@ -580,6 +580,7 @@ class Platform(StrEnum):
 IRRADIATION_BTUS_PER_HOUR_SQUARE_FOOT: Final = "BTU/(h×ft²)"
 
 # Precipitation units
+PRECIPITATION_MILLIMETERS: Final = "mm"
```

Comment:
Where is this used?

Reply:
Should be used in the weather platform for precipitation_unit
https://github.com/home-assistant/core/blob/dev/homeassistant/components/weather/__init__.py#L190

Reply:
But it isn't used?

As in, I don't see it being used anywhere, thus this would introduce a unused constant / dead code?

Reply:
It's being used here but to be honest I'm not exactly sure what this does.
https://github.com/home-assistant/core/blob/dev/homeassistant/components/weather/__init__.py#L287

Reply:
![image](https://user-images.githubusercontent.com/195327/149776510-03df7a35-c8fb-4ece-ab52-a2771378a2af.png)

I don't see `PRECIPITATION_MILLIMETERS` being used there at all?

Reply:
In the integration it would be `_attr_precipitation_unit = PRECIPITATION_MILLIMETERS`
Could also use the various length constants or simply leave it blank but I would prefer to set it if it's there.
Also there is one for inches used in various sensors (can't find the weather integration again where I saw it) so would it not make sense to have it also then for metric.

Reply:
In that case, I suggest adding it locally there. We make global constants when we have multiple integrations that use them.

Closing this PR for now.